src/ActivityLogger.php

```diff
@@ -2,129 +2,123 @@
+    public function __construct(Repository $config, ActivityLogStatus $logStatus, ActivityLoggerBatch $batch, CauserResolver $causerResolver)
     {
-        $this->auth = $auth;
+        $this->causerResolver = $causerResolver;
 
-        $this->authDriver = $config['activitylog']['default_auth_driver'] ?? $auth->getDefaultDriver();
+        $this->batch = $batch;
 
         $this->defaultLogName = $config['activitylog']['default_log_name'];
 
         $this->logStatus = $logStatus;
     }
 
-    public function setLogStatus(ActivityLogStatus $logStatus)
+    public function setLogStatus(ActivityLogStatus $logStatus): self
     {
         $this->logStatus = $logStatus;
 
         return $this;
     }
 
-    public function performedOn(Model $model)
+    public function performedOn(Model $model): self
     {
         $this->getActivity()->subject()->associate($model);
 
         return $this;
     }
 
-    public function on(Model $model)
+    public function on(Model $model): self
     {
         return $this->performedOn($model);
     }
 
-    public function causedBy($modelOrId)
+    public function causedBy(mixed $modelOrId): self
     {
         if ($modelOrId === null) {
             return $this;
         }
 
-        $model = $this->normalizeCauser($modelOrId);
+        $model = $this->causerResolver->resolve($modelOrId);
 
         $this->getActivity()->causer()->associate($model);
 
         return $this;
     }
 
-    public function by($modelOrId)
+    public function by(mixed $modelOrId): self
     {
         return $this->causedBy($modelOrId);
     }
 
-    public function causedByAnonymous()
+    public function causedByAnonymous(): self
     {
         $this->activity->causer_id = null;
         $this->activity->causer_type = null;
 
         return $this;
     }
 
-    public function byAnonymous()
+    public function byAnonymous(): self
     {
         return $this->causedByAnonymous();
     }
 
-    public function event(string $event)
+    public function event(string $event): self
     {
         return $this->setEvent($event);
     }
 
-    public function setEvent(string $event)
+    public function setEvent(string $event): self
     {
         $this->activity->event = $event;
 
         return $this;
     }
 
-    public function withProperties($properties)
+    public function withProperties(mixed $properties): self
     {
         $this->getActivity()->properties = collect($properties);
 
         return $this;
     }
 
-    public function withProperty(string $key, $value)
+    public function withProperty(string $key, mixed $value): self
     {
         $this->getActivity()->properties = $this->getActivity()->properties->put($key, $value);
 
         return $this;
     }
 
-    public function createdAt(Carbon $dateTime)
+    public function createdAt(Carbon $dateTime): self
     {
         $this->getActivity()->created_at = $dateTime;
```

Comment:
I know that this was Carbon before but I think that type-hinting the underlying interface and enforce carbon inside the method is more appropriate.
That way we also prevent a change of the `created_at` if the passed-in object is manipulated afterward as the `Carbon::instance()` uses `clone`.

```suggestion
    public function createdAt(DateTimeInterface $dateTime): self
    {
        $this->getActivity()->created_at = Carbon::instance($dateTime);
```